guvnor-ala/guvnor-ala-registry-vfs/src/main/java/org/guvnor/ala/registry/vfs/VFSPipelineExecutorRegistry.java

```diff
@@ -0,0 +1,116 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.guvnor.ala.registry.vfs;
+
+import java.util.List;
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.guvnor.ala.pipeline.execution.PipelineExecutorTrace;
+import org.guvnor.ala.registry.impl.BasePipelineExecutorRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.uberfire.java.nio.file.Path;
+
+import static org.guvnor.ala.registry.vfs.VFSRegistryHelper.BySuffixFilter.newFilter;
+import static org.uberfire.commons.validation.PortablePreconditions.checkNotNull;
+
+/**
+ * VFS based implementation of a PipelineExecutorRegistry.
+ */
+@ApplicationScoped
+public class VFSPipelineExecutorRegistry
+        extends BasePipelineExecutorRegistry {
+
+    protected static final String PIPELINE_EXECUTOR_REGISTRY_PATH = "executor-registry";
+
+    protected static final String TRACE_SUFFIX = "-trace.entry";
+
+    private static final Logger logger = LoggerFactory.getLogger(VFSPipelineExecutorRegistry.class);
+
+    private VFSRegistryHelper registryHelper;
+
+    private org.uberfire.java.nio.file.Path registryRoot;
+
+    public VFSPipelineExecutorRegistry() {
+        //Empty constructor for Weld proxying
+    }
+
+    @Inject
+    public VFSPipelineExecutorRegistry(final VFSRegistryHelper registryHelper) {
+        this.registryHelper = registryHelper;
+    }
+
+    @PostConstruct
+    public void init() {
+        initializeRegistryRoot();
+        initializeRegistry();
+    }
+
+    @Override
+    public void register(final PipelineExecutorTrace trace) {
+        checkNotNull("trace",
+                     trace);
+        final Path path = buildTracePath(trace.getTaskId());
+        try {
+            registryHelper.storeEntry(path,
+                                      trace);
+        } catch (Exception e) {
+            //uncommon error
+            logger.error("Unexpected error was produced during trace marshalling/storing, trace: " + trace,
+                         e);
+            throw new RuntimeException("Unexpected error was produced during trace marshalling/storing, trace: " + trace,
+                                       e);
+        }
+        super.register(trace);
+    }
+
+    public void deregister(final String taskId) {
+        checkNotNull("taskId",
+                     taskId);
+        final Path path = buildTracePath(taskId);
+        registryHelper.deleteBatch(path);
+        super.deregister(taskId);
+    }
+
+    private void initializeRegistryRoot() {
+        try {
+            registryRoot = registryHelper.ensureDirectory(PIPELINE_EXECUTOR_REGISTRY_PATH);
+        } catch (Exception e) {
+            //uncommon error
+            logger.error("An error was produced during " + VFSPipelineExecutorRegistry.class.getName() +
```

Comment:
Well, see that this fails typically in cases when the application "started" and the "VFS in general is broken".
This type or errors are let's say uncomon but also AFAIK there's no way to do let's say from a particular component a sort of application global status setup for setting the application in an application global error status. In general when  the VFS fails the application starts failing in different places, e.g. the Preferences can't be read, the initial indexing will fail, etc. So in this context of a failing VFS I believe it's ok to let this component just log that the it was not possible to initialize this directory.